openwebstart/src/main/java/com/openwebstart/jvm/ui/dialogs/ConfigurationDialog.java

```diff
@@ -99,4 +119,29 @@ public ConfigurationDialog() {
         add(panel);
     }
 
+    private static URL getUrl(String text) {
+        try {
+            return new URL(text);
+        } catch (MalformedURLException e) {
+            return null;
+        }
+    }
+
+    private String[] getVendorNames(URL specifiedServerURL) {
+        SwingWorker<String[], Void> swingWorker = new SwingWorker<String[], Void>() {
+            @Override
+            protected String[] doInBackground() {
+                LOG.info("Getting Vendor names");
+                return JavaRuntimeManager.getAllVendors(specifiedServerURL);
+            }
+        };
+        swingWorker.execute();
+        try {
+            return swingWorker.get(10, TimeUnit.SECONDS);
```

Comment:
In this line you block the EDT for maybe 10 seconds. As mentioned in the JavaDoc of the get method:
> calling {@code get} on the <i>Event Dispatch Thread</i> blocks <i>all</i> events, including repaints, from being processed until this {@code SwingWorker} is complete.
Please do async updates of the vendor model.

Reply:
SwingWorker could be a solution if it is used in a correct way. To be true since Java 8 the SwiongWorker do not make much sense anymore. By using Executors and Lambdas the code can be written in a much easier / better readable way.

Reply:
Example:

Executor.submit(() -> {
final List<String> vendors = loadVendorData();
SwingUtilities.invokeLater(() -> setVendorsToUi(vendors));
})

In this sample exception handling is missing and we should show the background work / progress in the UI.

Reply:
You can find some additional info at this slide deck (starting at slide 12): https://www.slideshare.net/HendrikEbbers/javafx-enterprise-javaone-2014

Reply:
We need to wait to update the model till http request running on background thread  comes back . The question is how long do we wait if it is not coming back fast. Normally it happens within 1 sec. It is coded in the way that if Http request does not come back in 10 secs then we update the model anyway. So in the worst case we wait for <= 10 secs. We can adjust this amount. But I see your point. The parallel thread can update the model in SwingInvoke later.

Let's talk in the daily.

Reply:
That is not the point. We should never wait in the EDT. Not event 1 second or 500 ms. This is just bad practice when doing UI related code (like in Swing, JavaFX, Javascript...). The long running http call must be handled in a background thread and publish its result on the ui thread without blocking the UI thread.

See:
- https://en.wikipedia.org/wiki/Event_dispatching_thread
- https://docs.oracle.com/javase/tutorial/uiswing/concurrency/dispatch.html
- https://books.google.de/books?id=FRbVIYAimp0C&pg=PT70&lpg=PT70&dq=swing+block+EDT&source=bl&ots=Yq518nLCOQ&sig=ACfU3U3wN47eNcsyuBJSl8FQ5EKZrLDAFw&hl=de&sa=X&ved=2ahUKEwiuyoqklIzlAhXPGuwKHbFnBvkQ6AEwA3oECAgQBA#v=onepage&q=swing%20block%20EDT&f=false

Reply:
Hendrik I am aware of this and hence the use of SwingWorker.  Was not sure about the timeout of Http request is set - it is not set in our case =>  hence the use of SwingWorker get() with timeout. However, internally the socket may have a time out.

Reply:
A question I have is that you already have an executor in main panel. Should it be used by the dialog by passing it in arg or should dialog have its own?

Reply:
About executor: Until now we do not have a central executor. Would make sense to refactor the code in that direction but for now `Executors.newSingleThreadExecutor()` can be used.

Reply:
about swingworker: it is not important if the http request has a timeout or not. Calling swing worker.get() with or without timeout will block the EDT.